dbms/src/IO/Progress.cpp

```diff
@@ -8,66 +8,79 @@
 
 namespace DB
 {
-
-void ProgressValues::read(ReadBuffer & in, UInt64 /*server_revision*/)
+void AllProgressValueImpl::read(ProgressValues & value, ReadBuffer & in, UInt64 /*server_revision*/)
 {
     size_t new_rows = 0;
     size_t new_bytes = 0;
     size_t new_total_rows = 0;
-
+    size_t new_write_rows = 0;
+    size_t new_write_bytes = 0;
+ 
     readVarUInt(new_rows, in);
     readVarUInt(new_bytes, in);
     readVarUInt(new_total_rows, in);
-
-    rows = new_rows;
-    bytes = new_bytes;
-    total_rows = new_total_rows;
-}
-
-
-void ProgressValues::write(WriteBuffer & out, UInt64 /*client_revision*/) const
+    readVarUInt(new_write_rows, in);
+    readVarUInt(new_write_bytes, in);
+ 
+    value.rows = new_rows;
+    value.bytes = new_bytes;
+    value.total_rows = new_total_rows;
+    value.write_rows = new_write_rows;
+    value.write_bytes = new_write_bytes;
+} 
+
+
+void AllProgressValueImpl::write(const ProgressValues & value, WriteBuffer & out, UInt64 /*client_revision*/)
```

Comment:
Backward compatibility?

Reply:
Backward compatibility yes. First different implementation were made of this helpers because compared about how we count the read rows, it's not really on real time but only in the end of query. So it wouldn't made that much sense at the moment that we show this value during the progress.

So AllProgressValueImpl has been implemented as a default case as it's quite likely the one would need to use. While the other implementation (concerning the read and write progress), is quite a specific case as it's used when a client use an HTTP request.

Reply:
Wait I just think of, do you talk about backward compatibility with new version of database and old client ? this PR don't handle this case right now.

Reply:
Yes, we maintain compatibility between server and client (including drivers with native protocol) in both ways: old client with new server and new client with old server.

Reply:
Ok thanks, now it's fixed. I will just need to probably need to modify the file `Define.h` before the code got merged if we got an another release